src/io/flutter/run/LauncherState.java

```diff
@@ -96,7 +96,9 @@ private RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws Ex
     final FlutterApp app = callback.createApp(device);
 
     // Remember the run configuration that started this process.
-    app.getProcessHandler().putUserData(KEY, runConfig);
+    app.getProcessHandler().putUserData(FLUTTER_RUN_CONFIG_KEY, runConfig);
+    app.getProcessHandler().putUserData(FLUTTER_APP_KEY, app);
+    app.getProcessHandler().putUserData(FLUTTER_LAUNCH_MODE_KEY, getEnvironment().getExecutor().getId());
```

Comment:
Maybe better to add the launch mode as a field to the FlutterApp class? Then it's more generally available, and you only need two keys here.